### ISSUE TITLE
More option for grid density, so that Igx grid row can be made narrower, compact is not so compact. Preferably input for rowHeight.

### ISSUE THREAD

**Issue by amitmoondra** [IgniteUI/igniteui-angular#11596]
## Is your feature request related to a problem? Please describe.
We want Igx Grid row narrower. For us, compact is also not very compact.

This is very important to our clients as they want to see more data in the screen. But right now even with compact configuration, the data visible on the screen is very less.

## Describe the solution you'd like
Currently, you have the below options, we want a way in which we can provide row height as input to the grid.
[rowHeight] = 22px etc. or something, to make the rows narrower.
“compact”: 56px
“cosy”: 64px
“comfortable “: 80px

## Additional context
We tried adjusting the height of the row by CSS but that leads to the virtual scroll getting messed up, which is correct as the internal calculation must have been affected and still using compact configuration.

**Comment by zdrawku** [comment]
@amitmoondra, @ChronosSF is going to provide more info on the "super compact" density.

As for the [rowHeight property](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxGridComponent.html#rowHeight), we [do have](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxGridComponent.html#rowHeight) such of the igxGrid, as well as [summaryRowHeight ](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxGridComponent.html#summaryRowHeight) and [pinnedRowHeight](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxGridComponent.html#pinnedRowHeight)


[Here ](https://github.com/IgniteUI/igniteui-angular/blob/1825e043c7/projects/igniteui-angular/src/lib/grids/grid-base.directive.ts#L3859)you can find the default values.

**Comment by amitmoondra** [comment]
I tried to add rowHeight to the grid and then tried scrolling the grid when the column group was applied to the grid. By scrolling I mean holding down the scroll handle and then dragging.
You will see here in the example that the focus is lost on the scroll handle.

https://stackblitz.com/edit/angular-tywl1u-wchf8d?file=src%2Fapp%2Fgrid%2Fgrid-groupby-sample%2Fgrid-groupby-sample.component.html
Let me know if this is an issue or expected behavior, but scrolling using scrollbar experience is not good in this case. Also only happens when grouping is applied. Scrolling will be fine if you remove the column group.
We also recorded a video of the same.https://equitydatascience.zoom.us/rec/share/ftFJfZT-tUlJjVXSNr2TS15q-sOjlJ2GqfwIMogO1K3hKiRECobDlHmq9N9Q1FFb.7blwl-FiKxBAbqSW Passcode: +f7093ul

I feel this is a issue as i could recreate it.
